### PR TITLE
Added a note that gc might try to pull missing .dir cache files

### LINKED ISSUE #1075: cmd-ref: add a note that gc might try to pull missing .dir cache files
https://discordapp.com/channels/485586884165107732/563406153334128681/690239392094486641

### PR DESCRIPTION
**Pull request by imhardikj**
Fixes #1075 
Changes as per observations in https://github.com/iterative/dvc.org/issues/1075#issuecomment-643669420

### CODE REVIEW

File: content/docs/command-reference/gc.md [hunk 1]
```diff
@@ -42,6 +42,9 @@ most cases removing data locally and in remote storage is irreversible.
 The default remote is cleaned (see `dvc config core.remote`) unless the
 `--remote` option is used.
 
+> Please note that `dvc gc` might try to pull `.dir` <abbr>cache</abbr> files
```
Comment (shcheklein) [inline content/docs/command-reference/gc.md:45]: - This note applies to the general case - consider moving it up
- May be we should change the language from "might" to "pulls"?
- Consider including a link to the https://dvc.org/doc/user-guide/dvc-files-and-directories#structure-of-cache-directory - it contains information about `.dir` files.
- Consider being more explicit - it pulls them when they are missing and they are referenced in the commits GC analyzed in `.dvc` or `dvc.yaml` files.

File: content/docs/command-reference/gc.md [hunk 2]
```diff
@@ -20,14 +20,20 @@ DVC cache but are no longer needed. With `--cloud` it also removes data in
...
-(so that DVC can figure out what fils can be safely deleted).
+(so that DVC can figure out what files can be safely deleted).
 
 One of the scope options (`--workspace`, `--all-branches`, `--all-tags`,
 `--all-commits`) or a combination of them must be provided. Each of them
 corresponds to keeping the data for the current workspace, and for a certain set
 of commits (determined by reading the DVC-files in them). See the
 [Options](#options) section for more details.
 
+> Please note that `dvc gc` pulls `.dir` >
```
Comment (imhardikj) [inline content/docs/command-reference/gc.md:31]: @shcheklein pre commit hook is adding ">" at end of this line.

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md]: Hi. This is resolved right? Doesn't seem correct.

File: content/docs/command-reference/gc.md [hunk 3]
```diff
@@ -20,14 +20,20 @@ DVC cache but are no longer needed. With `--cloud` it also removes data in
...
 `--all-commits`) or a combination of them must be provided. Each of them
 corresponds to keeping the data for the current workspace, and for a certain set
 of commits (determined by reading the DVC-files in them). See the
 [Options](#options) section for more details.
 
+> Please note that `dvc gc` pulls
+> [`.dir` cache](/doc/user-guide/dvc-files-and-directories#structure-of-cache-directory)
+> files for corresponding directories from remote. It pulls them when they are
+> missing and referenced in the commits that `gc` analyzed in `.dvc` or
+> `dvc.yaml` files.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:35]: This note is directly about `dvc gc` so no need for the block quote (no `>`s please).

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md:35]: Actually no... It's kind of a very technical implementation detail so the block quotes can stay. Never mind!

Comment (jorgeorpinel) [inline content/docs/command-reference/gc.md:35]: > Note that `dvc gc` tries to fetch any missing [`.dir` files]{linked} from [remote storage]{linked} to the local \<abbr>cache\</abbr>, in order to know which files should exist inside cached directories. These files may be missing if the cache directory was previously garbage collected, in a newly cloned copy of the repo, etc.

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md:35]: I'm not sure about this part:

> It pulls them when they are missing and referenced in the commits that `gc` analyzed in `.dvc` or `dvc.yaml` files.

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md:35]: Is that trye @efiop ? ☝️  And also, is the operation performed recursively as deeply as needed? Thanks

File: content/docs/command-reference/gc.md (hunk 2 above)
Comment (imhardikj) [reply content/docs/command-reference/gc.md]: I have added `.dir` to next line. The problem seems with using backtick " ` " at end of line(in a note). Pre commit hook doesn't except this.

Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md]: Strange. Never saw that before. Good to know...

File: content/docs/command-reference/gc.md (hunk 3 above)
Comment (efiop) [reply content/docs/command-reference/gc.md]: @jorgeorpinel Yes, looks correct. 

> And also, is the operation performed recursively as deeply as needed? 

Not sure what you mean here.

**Comment by shcheklein** [comment]
Thanks : 🙏

File: content/docs/command-reference/gc.md (hunk 3 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/gc.md]: I meant: Does it pull .dir files recursively from cache? .dir files mapped inside .dir files and so on — as many as needed.

Comment (efiop) [reply content/docs/command-reference/gc.md]: > .dir files mapped inside .dir files and so on — as many as needed

That doesn't ever happen, so it doesn't.

**Pull request merged**